### ISSUE TITLE
Decode error when cat has a weird error - this blocks everything [ACD-728]

### ISSUE THREAD

**Issue by Rafal80** [10clouds/aiconsole#259]
Migrated from issue: [ACD-728](https://10clouds.atlassian.net/browse/ACD-728) (Bug)

2023-11-28 14:57:01,921 aiconsole.api.endpoints.run_code [DEBUG] Running code: cat ~/Desktop/*.pdf,VM5:2 Exception in thread Thread-2779 (handle_stream_output):,Traceback (most recent call last):,  File “/Applications/AIConsole.app/Contents/Resources/python/lib/python3.10/threading.py”, line 1016, in _bootstrap_inner,    self.run(),  File “/Applications/AIConsole.app/Contents/Resources/python/lib/python3.10/threading.py”, line 953, in run,    self._target(*self._args, **self._kwargs),  File “/Applications/AIConsole.app/Contents/Resources/python/lib/python3.10/site-packages/aiconsole/core/code_running/code_interpreters/subprocess_code_interpreter.py”, line 160, in handle_stream_output,    for line in iter(stream.readline, “”):,  File “/Applications/AIConsole.app/Contents/Resources/python/lib/python3.10/codecs.py”, line 322, in decode,    (result, consumed) = self._buffer_decode(data, self.errors, final),UnicodeDecodeError: ‘utf-8’ codec can’t decode byte 0xe2 in position 10: invalid continuation byte,s @ VM5:2,emit @ VM4 sandbox_bundle:2,onMessage @ VM4 sandbox_bundle:2,index-1fcd6d89.js:117 Uncaught (in promise) TimeoutError: Request timed out,    at index-1fcd6d89.js:117:5442,(anonymous) @ index-1fcd6d89.js:117,setTimeout (async),(anonymous) @ index-1fcd6d89.js:117,CEe @ index-1fcd6d89.js:117,_fetch @ index-1fcd6d89.js:117,a @ index-1fcd6d89.js:117,await in a (async),create @ index-1fcd6d89.js:117,t.<computed> @ index-1fcd6d89.js:117,QSe @ index-1fcd6d89.js:117,runApiWithProcess @ index-1fcd6d89.js:117,doRun @ index-1fcd6d89.js:117,Abe @ index-1fcd6d89.js:117,Mbe @ index-1fcd6d89.js:117,r.onmessage @ index-1fcd6d89.js:117,_handleMessage @ index-1fcd6d89.js:101,index-1fcd6d89.js:117 Uncaught (in promise) TimeoutError: Request timed out,    at index-1fcd6d89.js:117:5442,(anonymous) @ index-1fcd6d89.js:117,await in (anonymous) (async),(anonymous) @ index-1fcd6d89.js:117,r @ index-1fcd6d89.js:101,finishProcess @ index-1fcd6d89.js:117,Ibe @ index-1fcd6d89.js:117,Mbe @ index-1fcd6d89.js:117,r.onmessage @ index-1fcd6d89.js:117,_handleMessage @ index-1fcd6d89.js:101 

[ACD-728]: https://10clouds.atlassian.net/browse/ACD-728?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by pbaranski9411** [comment]
<img width="1081" alt="image" src="https://github.com/10clouds/aiconsole/assets/121805402/eb55c93c-e95c-4f7b-b14c-586fdfbb3564">

Hmm, I think it's working properly. Probably that was specific .pdf file which is readable, but always AIConsole is responding to me that way and don't proceed the request further.

I'm switching it to qa

**Comment by cielecki** [comment]
Bart can supply this pdf as it was on his macbook.

**Comment by cielecki** [comment]
He is on discord

**Comment by pbaranski9411** [comment]
Waiting for pdfs from Bart

**Comment by pbaranski9411** [comment]
Try replicate this error with polish symbols

**Comment by cielecki** [comment]
This should not be on me

**Comment by aga-gasinska** [comment]
force it to cut it anyway (try with polish signs in pdf)

**Comment by arudzinska** [comment]
As I'm giving away this issue, my current findings are:
- I tried to force the assistant to cat the file anyways, but out of three tries I think it listened to me only once (other two times it still complained it cannot cat PDF files). What happened then is that I got a different error that the one described in this issue (UnicodeDecodeError), but I don't remember what it was.
- I prepared two PDF files (one with and one without Polish symbols) that I used for tests and they are attached below.

[pdf-no-pl.pdf](https://github.com/10clouds/aiconsole/files/14031196/pdf-no-pl.pdf)
[pdf-with-pl.pdf](https://github.com/10clouds/aiconsole/files/14031197/pdf-with-pl.pdf)
